keras_nlp/models/whisper/whisper_audio_to_speech_lm.py

```diff
@@ -0,0 +1,498 @@
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Whisper Seq2Seq LM (Language Model)."""
+
+import copy
+import os
+
+import tensorflow as tf
+from tensorflow import keras
+
+from keras_nlp.api_export import keras_nlp_export
+from keras_nlp.models.generative_task import GenerativeTask
+from keras_nlp.models.whisper.whisper_audio_to_speech_lm_preprocessor import (
+    WhisperAudioToSpeechLMPreprocessor,
+)
+from keras_nlp.models.whisper.whisper_backbone import WhisperBackbone
+from keras_nlp.models.whisper.whisper_presets import backbone_presets
+from keras_nlp.utils.keras_utils import is_xla_compatible
+from keras_nlp.utils.python_utils import classproperty
+from keras_nlp.utils.python_utils import format_docstring
+
+
+@keras_nlp_export("keras_nlp.models.WhisperAudioToSpeechLM")
+class WhisperAudioToSpeechLM(GenerativeTask):
+    """An end-to-end Whisper model for seq2seq language modeling.
+
+    A seq2seq language model (LM) is an encoder-decoder model which is used for
+    conditional text generation. The encoder is given a "context" text (fed to
+    the encoder), and the decoder predicts the next token based on both the
+    encoder inputs and the previous tokens. You can finetune `WhisperAudioToSpeechLM` to
+    generate text for any seq2seq task (e.g., translation or summarization).
+
+    This model has a `generate()` method, which generates text based on
+    encoder inputs and an optional prompt for the decoder. The generation
+    strategy used is controlled by an additional `sampler` argument passed to
+    `compile()`. You can recompile the model with different `keras_nlp.samplers`
+    objects to control the generation. By default, `"top_k"` sampling will be
+    used.
+
+    This model can optionally be configured with a `preprocessor` layer, in
+    which case it will automatically apply preprocessing to string inputs during
+    `fit()`, `predict()`, `evaluate()` and `generate()`. This is done by default
+    when creating the model with `from_preset()`.
+
+    Disclaimer: Pre-trained models are provided on an "as is" basis, without
+    warranties or conditions of any kind. The underlying model is provided by a
+    third party and subject to a separate license, available
+    [here](https://github.com/facebookresearch/fairseq/).
+
+    Args:
+        backbone: A `keras_nlp.models.WhisperBackbone` instance.
+        preprocessor: A `keras_nlp.models.WhisperAudioToSpeechLMPreprocessor` or `None`.
+            If `None`, this model will not apply preprocessing, and inputs
+            should be preprocessed before calling the model.
+    """
```

Comment:
Given that we have presets, we can probably go ahead and add docstrings. Also keep in mind that after the multi-backend changes, we will favor `np.array` over `tf.constant`.